src/client/clientobject.cpp

```diff
@@ -55,6 +57,30 @@ ClientActiveObject* ClientActiveObject::create(ActiveObjectType type,
 	return object;
 }
 
+
+void ClientActiveObject::selectIfInRange(const Shootline &shootline,
+		std::vector<DistanceSortedActiveObject> &dest)
+{
+	// Imagine a not-axis-aligned cuboid oriented into the direction of the shootline,
+	// with the width of the object's selection box radius * 2 and with length of the
+	// shootline (+selection box radius forwards and backwards). We check whether
+	// the selection box center is inside this cuboid.
+	aabb3f selection_box;
+	if (!getSelectionBox(&selection_box))
+			return;
+
+	// possible optimization: get rid of the sqrt here
+	f32 selection_box_radius = selection_box.getRadius();
+
+	v3f pos_diff = getPosition() + selection_box.getCenter() - shootline.getLine().start;
+
+	f32 d = shootline.getDir().dotProduct(pos_diff);
+
+	if (shootline.hitsSelectionBox(pos_diff, d, selection_box_radius)) {
```

Comment:
Instead of passing `pos_diff` and `d`, I’d pass `getPosition() + selection_box.getCenter()` and return `d`.

Reply:
What value would you use to indicate falsyness?

Reply:
`std::nullopt` likely, it’s allowed now (since last Sunday)

Reply:
Ah, std::optional from C++17. Very good!